### PR TITLE
Add node version check

### PR DESCRIPTION
**Pull request by sebrandon1**
Notable changes as part of this PR:
- Added automated RHCOS version mapping script that will create a PR with any changes.
- Added logic surrounding our new platform test `ocp-node-os-lifecycle`.  Checks each node for compatibility between OCP and available RHEL/RHCOS workers.
- Added `projectpath` package which helps the ability to reference text files between different packages.
- Catalog entry added.
- Added make entry `make update-rhcos-versions` which will kick off the script to load the latest shortVersion / longVersion into our cache file for RHCOS versions.  Based on script found here: https://access.redhat.com/articles/5449111

### CODE REVIEW

File: .github/workflows/update-rhcos-mapping.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Update RHCOS version mapping
+
+on:
+  # Triggers the workflow every day
```
[the remark this answers is not in the document]
Comment (sebrandon1) [reply .github/workflows/update-rhcos-mapping.yml:4]: I think if the file doesn't change, it doesn't create the PR.

File: script/rhcos_versions.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+rm -f ./rhcos_version_map &> /dev/null
+
+for i in ${CHANNELS[@]}; do
+    VERSIONS=$(curl -sH 'Accept: application/json'  "https://api.openshift.com/api/upgrades_info/v1/graph?channel=stable-${i}" | jq '.nodes[].version' -r)
+
+    for VERSION in ${VERSIONS}
+    do
+        # Look up the release version using oc adm.
+        RHCOSVERSION="$(oc adm release info "${VERSION}" -o 'jsonpath={.displayVersions.machine-os.Version}')"
+        if [ ! -z ${RHCOSVERSION} ]; then
```
Comment (rdavid) [inline script/rhcos_versions.sh:16]: This could failed as well: cluster is down or does not answer.

File: script/rhcos_versions.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            echo "$VERSION / $RHCOSVERSION" | tee -a ./rhcos_version_map
+        fi
+    done
+done
+
+sort -o ./rhcos_version_map ./rhcos_version_map 
+
+echo OpenShift to RHCOS version mapping is in 'rhcos_version_map'
```
Comment (rdavid) [inline script/rhcos_versions.sh:24]: No need for single quotes.

File: script/rhcos_versions.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+for i in ${CHANNELS[@]}; do
+    VERSIONS=$(curl -sH 'Accept: application/json'  "https://api.openshift.com/api/upgrades_info/v1/graph?channel=stable-${i}" | jq '.nodes[].version' -r)
+
+    for VERSION in ${VERSIONS}
+    do
+        # Look up the release version using oc adm.
+        RHCOSVERSION="$(oc adm release info "${VERSION}" -o 'jsonpath={.displayVersions.machine-os.Version}')"
+        if [ ! -z ${RHCOSVERSION} ]; then
+            echo "$VERSION / $RHCOSVERSION" | tee -a ./rhcos_version_map
```
Comment (rdavid) [inline script/rhcos_versions.sh:17]: Do you clean this temporary file somewhere? Could be cleaned before the cycle.

Comment (sebrandon1) [reply script/rhcos_versions.sh:17]: Yes, above I do: `rm -f ./rhcos_version_map &> /dev/null`

File: script/rhcos_versions.sh (hunk 3 above)
Comment (sebrandon1) [reply script/rhcos_versions.sh:24]: Fixed.  This was originally copied from: https://access.redhat.com/articles/5449111

File: script/rhcos_versions.sh (hunk 2 above)
Comment (sebrandon1) [reply script/rhcos_versions.sh:17]: Covered by the set -e.

File: script/rhcos_versions.sh [hunk 5]
```diff
@@ -21,4 +22,4 @@ done
 
 sort -o ./rhcos_version_map ./rhcos_version_map 
 
-echo OpenShift to RHCOS version mapping is in 'rhcos_version_map'
+echo "OpenShift to RHCOS version mapping is in rhcos_version_map"
```
Comment (rdavid) [inline script/rhcos_versions.sh:25]: No need for any quotation. Shell is very string friendly.

**Review by rdavid — CHANGES_REQUESTED** [review]

File: script/rhcos_versions.sh (hunk 5 above)
Comment (sebrandon1) [reply script/rhcos_versions.sh]: Done.

File: script/rhcos_versions.sh [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+set -x
+
+CHANNELS=("4.10" "4.9" "4.8" "4.7" "4.6" "4.5" "4.4" "4.3" "4.2" "4.1")
+
+rm -f ./rhcos_version_map &> /dev/null
+
+for i in ${CHANNELS[@]}; do
+    VERSIONS=$(curl -sH 'Accept: application/json'  "https://api.openshift.com/api/upgrades_info/v1/graph?channel=stable-${i}" | jq '.nodes[].version' -r)
+
```
[the remark this answers is not in the document]
Comment (sebrandon1) [reply script/rhcos_versions.sh:12]: Just going to handle this with `set -e` I believe.

File: cnf-certification-test/identifiers/identifiers.go [hunk 7]
```diff
@@ -358,6 +363,15 @@ func GetSuiteAndTestFromIdentifier(identifier claim.Identifier) []string {
 // Catalog is the JUnit testcase catalog of tests.
 var Catalog = map[claim.Identifier]TestCaseDescription{
 
+	TestNodeOperatingSystemIdentifier: {
+		Identifier: TestNodeOperatingSystemIdentifier,
+		Type:       normativeResult,
+		Description: formDescription(TestNodeOperatingSystemIdentifier, `Tests that the nodes running in the cluster have operating systems
+			that are compatible with the deployed version of OpenShift.`),
+		Remediation:           `Please update your workers to a version that is supported by your version of OpenShift`,
+		BestPracticeReference: bestPracticeDocV1dot3URL + " Section x.x", // TODO: Update this with correct section of requirements doc.
```
[the remark this answers is not in the document]
Comment (sebrandon1) [reply cnf-certification-test/identifiers/identifiers.go]: Done.

**Review by rdavid — APPROVED** [review]

**Pull request merged**